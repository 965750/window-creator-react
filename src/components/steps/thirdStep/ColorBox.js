import React from 'react'
import styles from './ColorBox.module.scss'
import { changeColorBox } from '../../../store/actions/creatorActions'
import { connect } from 'react-redux'

import BaseCheckbox from '../../blocks/BaseCheckbox'
import { FormattedMessage } from 'react-intl'

const ColorBox = ({ data, changeColorBox, classes }) => {
    return (
        <div
            className={`${classes} cursor-pointer`}
            onClick={() => changeColorBox(data.id)}
        >
            <div
                className={`${styles.colorBox} mx-auto rounded-full mb-1`}
                style={{ background: data.color }}
            />
            <div className="flex justify-center">
                <BaseCheckbox classes="mt-sm" isChecked={data.active} />
                <span className="ml-1">
                    <FormattedMessage
                        id={data.label}
                        defaultMessage={data.label}
                    />
                </span>
            </div>
        </div>
    )
}

const mapDispatchToProps = dispatch => {
    return {
        changeColorBox: id => dispatch(changeColorBox(id)),
    }
}

export default connect(
    null,
    mapDispatchToProps
)(ColorBox)
