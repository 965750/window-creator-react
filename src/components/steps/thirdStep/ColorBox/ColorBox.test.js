import '@testing-library/jest-dom/extend-expect'

import { applyMiddleware, compose, createStore } from 'redux';
import { IntlProvider } from 'react-intl'
import { Provider } from 'react-redux'
import { render, cleanup } from '@testing-library/react'

import ColorBox from './index'
import messages from '../../../../messages-i18n'
import thunk from 'redux-thunk'
import React from 'react'

const initialStore = {
  theme: {
    lang: 'pl',
  },
}

const reducer = (state = initialStore, action) => {
  switch (action.type) {
    default:
      return state
  }
}

const store = createStore(
  reducer,
  compose(
    applyMiddleware(thunk),
  ),
)

const renderWithRedux = (component) => ({
  ...render(<Provider store={store}>{component}</Provider>),
})

describe('ColorBox', () => {
  let wrapper
  const initialProps = {
    data: {
      id: 'mockId',
      color: 'red',
      active: false,
      label: 'Black'
    }
  }

  const setUp = (props = initialProps) => {
    wrapper = renderWithRedux(
      <IntlProvider
        locale="en"
        messages={messages[store.getState().theme.lang]}
      >
        <ColorBox {...props} />
      </IntlProvider>,
    )
  }

  afterEach(cleanup)

  it('should match snapshot', () => {
    setUp()

    expect(wrapper.asFragment()).toMatchSnapshot()
  })

  it('should be checked when is active has been passed', () => {
    setUp({
      ...initialProps,
      data: {
        ...initialProps.data,
        active: true,
      }
    })

    expect(wrapper.asFragment()).toMatchSnapshot()
  })

  it('should pass classes from props to wrapper', () => {
    const mockClasses = 'mockClasses'

    setUp({
      ...initialProps,
      classes: mockClasses,
    })

    expect(wrapper.queryByTestId('colorBoxWrapper')).toHaveClass(mockClasses)
  })
})
