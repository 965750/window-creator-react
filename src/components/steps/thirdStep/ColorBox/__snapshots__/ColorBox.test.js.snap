// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorBox should be checked when is active has been passed 1`] = `
<DocumentFragment>
  <button
    class="undefined cursor-pointer"
    data-testid="colorBoxWrapper"
  >
    <div
      class="colorBox mx-auto rounded-full mb-1"
      style="background: red;"
    />
    <div
      class="flex justify-center"
    >
      <div
        class="radio__wrapper border-botticelli border-4 rounded-full relative mt-sm"
        data-testid="radioWrapper"
      >
        <div
          class="radio--checked top-half left-half rounded-full bg-hippieBlue absolute"
        />
      </div>
      <span
        class="ml-1"
      >
        Czarny
      </span>
    </div>
  </button>
</DocumentFragment>
`;

exports[`ColorBox should match snapshot 1`] = `
<DocumentFragment>
  <button
    class="undefined cursor-pointer"
    data-testid="colorBoxWrapper"
  >
    <div
      class="colorBox mx-auto rounded-full mb-1"
      style="background: red;"
    />
    <div
      class="flex justify-center"
    >
      <div
        class="radio__wrapper border-botticelli border-4 rounded-full relative mt-sm"
        data-testid="radioWrapper"
      />
      <span
        class="ml-1"
      >
        Czarny
      </span>
    </div>
  </button>
</DocumentFragment>
`;
